### PR TITLE
Monorepo: Fix Mixer deps and add mixer to pilot e2e tests

### PR DESCRIPTION
**Pull request by mandarjog**
**What this PR does / why we need it**:
Fixes Mixer Segv by using only one prometheus dep in the binary.

```release-note
NONE
```

### CODE REVIEW

File: pilot/bin/push-docker [hunk 1]
```diff
@@ -72,10 +72,19 @@ bazel build ${BUILD_FLAGS} //pilot/cmd/... //pilot/test/...
...
+MIXER_LOCAL_IMG="istio/mixer/docker:mixer"
+
 pushd docker
-  for image in app proxy proxy_init proxy_debug pilot sidecar_initializer eurekamirror; do
-    local_image="${image}:${local_tag}"
-    docker build -q -f "Dockerfile.${image}" -t "${local_image}" .
+  for image in app proxy proxy_init proxy_debug pilot sidecar_initializer eurekamirror mixer; do
+    if [[ $image == "mixer" ]];then
+      # build mixer docker image, it is tagged as istio/mixer/docker:mixer
+      bazel run //mixer/docker:mixer
```
Comment (rshriram) [inline pilot/bin/push-docker]: please add BUILD_FLAGS to all bazel commands so as to allow building on mac.

File: mixer/adapter/prometheus/prometheus_test.go [hunk 2]
```diff
@@ -358,7 +358,7 @@ func TestProm_Record(t *testing.T) {
 						continue
 					}
 				case *prometheus.HistogramVec:
-					if err := c.(*prometheus.HistogramVec).With(promLabels(adapterVal.Dimensions)).Write(m); err != nil {
+					if err := c.(*prometheus.HistogramVec).With(promLabels(adapterVal.Dimensions)).(prometheus.Metric).Write(m); err != nil {
```
Comment (mandarjog) [inline mixer/adapter/prometheus/prometheus_test.go]: api changed on v08 to the current prometheus version.

File: mixer/example/servicegraph/promgen/promgen.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/prometheus/client_golang/api/prometheus"
+	prometheus "github.com/prometheus/client_golang/api/prometheus"
```
Comment (rkpagadala) [inline mixer/example/servicegraph/promgen/promgen.go]: is this change needed, looks like a no-op
